components/builder-jobsrv/src/data_store.rs

```diff
@@ -154,13 +154,34 @@ impl DataStore {
                                 END
                                 $$ LANGUAGE plpgsql VOLATILE"#)?;
 
+        migrator.migrate("jobsrv",
+                         r#"CREATE OR REPLACE FUNCTION reset_jobs_v1 () RETURNS SETOF jobs AS
+                                $$
+                                DECLARE
+                                    r jobs % rowtype;
+                                BEGIN
+                                    FOR r IN
+                                        SELECT * FROM jobs
+                                        WHERE job_state = 'Dispatched'
+                                        ORDER BY created_at ASC
+                                        FOR UPDATE SKIP LOCKED
+                                    LOOP
+                                        UPDATE jobs SET job_state='Pending', updated_at=now() WHERE id=r.id RETURNING * INTO r;
+                                        RETURN NEXT r;
+                                    END LOOP;
+                                  RETURN;
+                                END
+                                $$ LANGUAGE plpgsql VOLATILE"#)?;
```

Comment:
Can we just do this in a plain SQL function with an `UPDATE` statement without the looping?